spectator_mode/init.lua

```diff
@@ -129,27 +120,50 @@ local function hide_player(player)
 
 	player:set_properties(prop)
 	player:set_armor_groups({ immortal = 1 })
-	player:set_nametag_attributes({color = {a = 0, r = 255, g = 255, b = 255}})
+	player:set_nametag_attributes({color = {a = 0, r = 255, g = 255, b = 255}, text = ""})
 end
 
 minetest.register_on_joinplayer(function(player)
-	minetest.after(3, function(name)
-		local player = minetest.get_player_by_name(name)
-		if not player then
-			return
-		end
-
-		if not minetest.check_player_privs(name, { spectate = true }) then
-			return
-		end
+	if not minetest.check_player_privs(player:get_player_name(), { spectate = true }) then
+		return
+	end
 
-		player:set_armor_groups({immortal = 1})
-		old_set(player, ctf_playertag.TYPE_BUILTIN, { a=0, r=255, g=255, b=255 })
+	player:set_armor_groups({immortal = 1})
+	old_set(player, ctf_playertag.TYPE_BUILTIN, { a=0, r=255, g=255, b=255 })
```

Comment:
This should go into `hide_player`
`old_set` should replace `player:set_nametag_attributes` if they both work, remember to add the `text = ""` bit though